utils/isOwner.js

```diff
@@ -0,0 +1,20 @@
+const isOwner = async (collection, userId, id) => {
+  try {
+    const singleCollection = await collection.findOne({ id });
+
+    if(!singleCollection) return strapi.errors.badRequest('A record with that id was not found');
+
+    if(!singleCollection.user?.id) return strapi.errors.badRequest('This collection does not have a user id');
+
+    if(userId === singleCollection.user.id){
+      return true;
+    }
+
+    return false;
+  } catch(err) {
+    strapi.errors.badRequest(err);
```

Comment:
You already catch bad user inputs in the try branch. I think errors that reach the catch branch are internal errors, so semantically it's better to return status 500 here.